### PR TITLE
Added documention for ML on Linera

### PR DESCRIPTION
**Pull request by christos-h**
Added documentation for running machine learning models on Linera with candle.

This should suffice to get people started with ML on Linera - however given that ML inference on different models has a completely different process, we recommend that individuals consult the candle examples and docs directly.

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -37,6 +37,10 @@
   - [Creating New Blocks](advanced_topics/block_creation.md)
   - [Applications that Handle Assets](advanced_topics/assets.md)
 
+- [Experimental]()
```
Comment (christos-h) [inline src/SUMMARY.md:40]: Add link

File: src/experimental/ml.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
+# Machine Learning on Linera
+
+The Linera application contract / service split allows for securely and    |
```
Comment (christos-h) [inline src/experimental/ml.md:3]: Remove pipe

File: src/experimental/ml.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+`candle` is a minimalist ML framework for Rust with a focus on performance and
+usability. It also compiles to Wasm and has great support for Wasm both in 
+and outside the browser. Check candle's [examples](https://github.com/huggingface/candle/tree/main/candle-wasm-examples) 
+for inspiration on the types of models which are supported.
+
+### Getting Started
+
+To add ML capabilities to your existing Linera project, you'll need to add the `candle-core`, `getrandom`, `rand`
+and `tokenizers` dependencies to your Linera project:
```
Comment (christos-h) [inline src/experimental/ml.md:25]: Tokenizers so depend on transformers

**Review by ma2bd — APPROVED** [review]

File: src/experimental/ml.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        let raw_weights = self.runtime.fetch_url("https://my-model-provider.com/model.bin");
+        // do more stuff here
+    }
+}
+```
+
+This can be served from a local webserver or pulled directly from a model
+provider such as Hugging Face.
+
+At this we have the raw bytes which correspond to the models and tokenizer.
```
Comment (afck) [inline src/experimental/ml.md:95]: At this… point?

File: src/experimental/ml.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+[Blob API](https://github.com/linera-io/linera-protocol/issues/1981). The Blob
+API will enable large binary blobs to be stored on-chain, the correctness and
+availability of which is guaranteed by the validators.
+
+### Maximum Model Size
+
+The maximum size of a model which can be loaded into an application's service is
+currently constrained by:
+
+1. The addressable memory of the service's Wasm runtime being 4 Gb.
```
Comment (afck) [inline src/experimental/ml.md:150]:
```suggestion
1. The addressable memory of the service's Wasm runtime being 4 GiB.
```

File: src/experimental/ml.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
+# Machine Learning on Linera
+
+The Linera application contract / service split allows for securely and
+efficiently running machine learning models on the edge.
+
+The application's contract retrieves the correct model with all the correctness
```
Comment (afck) [inline src/experimental/ml.md:6]: Isn't it the service that retrieves the model, using `fetch_url`?

Comment (christos-h) [reply src/experimental/ml.md:6]: Yes, for now. And it's mentioned in the limitations section - however this is the longer term plan and I thought it was important to articulate it. I wonder how I can phrase this better.

**Pull request merged**